builder/dockerfile/dispatchers.go

```diff
@@ -317,7 +316,10 @@ func run(b *Builder, args []string, attributes map[string]bool, original string)
 
 	// stash the cmd
 	cmd := b.runConfig.Cmd
-	runconfig.Merge(b.runConfig, config)
+	if b.runConfig.Entrypoint.Len() == 0 && b.runConfig.Cmd.Len() == 0 {
+		b.runConfig.Cmd = config.Cmd
```

Comment:
Yes. config is created (L312) with a Cmd and Image, nothing more. Merge() doesnt mutate Image, only Cmd. So we are only merging Cmd here.